std/math.d

```diff
@@ -2371,6 +2391,8 @@ L_was_nan:
     assert(feqrel(exp2(0.5L), SQRT2) >= real.mant_dig -1);
     assert(exp2(8.0L) == 256.0);
     assert(exp2(-9.0L)== 1.0L/512.0);
+
+    static assert(exp2(-2.0L) == 0.25L); // CTFE
```

Comment:
... erroring out due to non-CTFE-able `floorImpl()`:

```
std/math.d(439): Error: y.vu[4] is used before initialized
std/math.d(412):        originally uninitialized here
std/math.d(4096):        called from here: floorImpl(x)
std/math.d(2372):        called from here: floor(x + 0.5L)
std/math.d(2109):        called from here: exp2Impl(x)
std/math.d(2395):        called from here: exp2(-2L)
std/math.d(2395):        while evaluating: `static assert(exp2(-2L) == 0.25L)`
```

So it'd be great if someone else continued on extending upstream std.math CTFE-ability. ;)